In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('worldcup2014.csv')

## Desafío 1:

### Primeras 5 observaciones de la copa 2014

In [3]:
df.head(5)

team     continent group  group_pos  cantidad_juegos  \
0       Brazil  southamerica     A          1                3   
1       Mexico  northamerica     A          2                3   
2      Croatia        europe     A          3                3   
3     Cameroon        africa     A          4                3   
4  Netherlands        europe     B          1                3   

   juegos_ganados  juegos_empatados  juegos_perdidos  goles_favor  \
0               2                 1                0            7   
1               2                 1                0            4   
2               1                 0                2            6   
3               0                 0                3            1   
4               3                 0                0           10   

   goles_contra goles_diferencia  puntos  clasificado  
0             2                5       7            1  
1             1                3       7            1  
2             6                0       3            0  
3             9               −8       0            0  
4             3                7       9            1

## Desafío 2:

### Las frecuencias de continent:

In [4]:
df['continent'].value_counts()

europe          13
southamerica     6
africa           5
northamerica     4
asia             4
Name: continent, dtype: int64

### El continente con una mayor presencia en la muestra es:
Europa

In [5]:
df['continent'].value_counts().idxmax()

'europe'

### La probabilidad de elegir un equipo asiático al azar es: 12.5 porciento

In [6]:
asia_probability = df['continent'].value_counts('%')['asia']
print(asia_probability * 100, "%")

12.5 %


### La probabilidad de elegir un equipo africano al azar es: 15.62 porciento

In [7]:
africa_probability = df['continent'].value_counts('%')['africa']
print(africa_probability * 100, "%")

15.625 %


## Desafío 3:

Por cada continente, genere un nuevo objeto que almacene sólo las observaciones del continente.

In [8]:
europe_df = df[df['continent'] == 'europe']
southamerica_df = df[df['continent'] == 'southamerica']
africa_df = df[df['continent'] == 'africa']
northamerica_df = df[df['continent'] == 'northamerica']
asia_df = df[df['continent'] == 'asia']

continent_observations = {
    'Europa': europe_df,
    'Sur América': southamerica_df,
    'África': africa_df,
    'Norte América': northamerica_df,
    'Asia': asia_df
}

## Desafío 4: Calcule la probabilidad de clasificación a la siguiente ronda

In [9]:
for continent, observations in continent_observations.items():
    classified_events_per_continent = observations['clasificado'].value_counts().get(1)
    if classified_events_per_continent != None:
        classification_probability_per_continent = (classified_events_per_continent / len(df)) * 100
    else:
        classification_probability_per_continent = 0
        
    print("La posibilidad de clasificar a la siguiente ronda de", continent, "es:", classification_probability_per_continent, "%")


    

La posibilidad de clasificar a la siguiente ronda de Europa es: 18.75 %
La posibilidad de clasificar a la siguiente ronda de Sur América es: 15.625 %
La posibilidad de clasificar a la siguiente ronda de África es: 6.25 %
La posibilidad de clasificar a la siguiente ronda de Norte América es: 9.375 %
La posibilidad de clasificar a la siguiente ronda de Asia es: 0 %


### ¿Cuál fue la probabilidad de que un país asiático pase a la siguiente ronda?:

La probabilidad para Asia es 0 porciento

### ¿Cuáles fueron los dos continentes que tuvieron la mayor probabilidad de clasificar?

Los dos continentes que tuvieron la mayor probabilidad de clasificar
fueron **Europa** con un porcentaje de 18.75 % y **Sur América** con un porcentaje de 15.625%

### ¿Cuál fue la probabilidad de no clasificar un país europeo?

In [10]:
continent_observations['Europa']['clasificado'].value_counts()

0    7
1    6
Name: clasificado, dtype: int64

In [11]:
(7/len(df)) * 100

21.875

La probabilidad de no clasificar a una sgte ronda de un país europeo es de 21.8 %

## Desafío 5: Refactorización

In [12]:
for d in [africa_df, europe_df, asia_df, northamerica_df,southamerica_df]:
    d_serie = pd.Series(np.where(d['juegos_ganados'] > 0, 1, 0))
    print(d_serie.value_counts('%'))

1    0.6
0    0.4
dtype: float64
1    0.846154
0    0.153846
dtype: float64
0    1.0
dtype: float64
1    0.75
0    0.25
dtype: float64
1    1.0
dtype: float64


In [13]:
for d in [africa_df, europe_df, asia_df, northamerica_df,southamerica_df]:
    d_serie = pd.Series(np.where(d['juegos_ganados'] > 0, 1, 0))
    print(d_serie.value_counts('%'))

1    0.6
0    0.4
dtype: float64
1    0.846154
0    0.153846
dtype: float64
0    1.0
dtype: float64
1    0.75
0    0.25
dtype: float64
1    1.0
dtype: float64


### De acuerdo a la totalidad de juegos, ¿Qué continente tuvo una mayor probabilidad de ganar juegos?

In [14]:
print("La totalidad de juegos es ", len(df))

La totalidad de juegos es  32


El continente con mayor probabilidad de ganar juegos es Sur América

### ¿Qué continente presentó un nivel similar entre juegos ganados y perdidos?

Africa fue el continente que presentó un nivel similar entre juegos ganados y perdidos

### Utilize el  código para el  caso de  juegos perdidos y  analize el  continente con una mayor probabilidad de perder.

El continente con mayor probabilidad de perder es Asia, debido a que ha perdido todos sus partidos